tests/markdown/check-markdown.test.js

```diff
@@ -0,0 +1,140 @@
+        mockConsoleLog.mockRestore();
+    });
+
+    it('returns multiple validation errors for invalid blog frontmatter', async () => {
+        const frontmatter = {
+            title: 123,
+            date: 'invalid-date',
+            type: 'blog',
+            tags: 'not-an-array',
+            cover: ['not-a-string'],
+            authors: { name: 'John Doe' }
+        };
+        const errors = validateBlogs(frontmatter);
+
+        expect(errors).toEqual([
+            "Invalid date format: invalid-date",
+            "Tags should be an array",
+            "Cover must be a string",
+            "Authors should be an array"]);
+    });
+
+    it('logs error to console when an error occurs in checkMarkdownFiles', async () => {
+        const invalidFolderPath = path.join(tempDir, 'non-existent-folder');
+
+        await expect(checkMarkdownFiles(invalidFolderPath, validateBlogs))
+            .rejects.toThrow('ENOENT');
+
+        expect(mockConsoleError.mock.calls[0][0]).toContain('Error in directory');
+    });
+
+    it('skips the "reference/specification" folder during validation', async () => {
+        const referenceSpecDir = path.join(tempDir, 'reference', 'specification');
+        await fs.mkdir(referenceSpecDir, { recursive: true });
+        await fs.writeFile(path.join(referenceSpecDir, 'skipped.md'), `---\ntitle: Skipped File\n---`);
+
+        const mockConsoleLog = jest.spyOn(console, 'log').mockImplementation();
+
+        await checkMarkdownFiles(tempDir, validateDocs);
+
+        expect(mockConsoleLog).not.toHaveBeenCalledWith(expect.stringContaining('Errors in file reference/specification/skipped.md'));
+        mockConsoleLog.mockRestore();
+    });
+
+    it('logs and rejects when an exception occurs while processing a file', async () => {
+        const filePath = path.join(tempDir, 'invalid.md');
+        await fs.writeFile(filePath, `---\ntitle: Valid Title\n---`);
+
+        const mockReadFile = jest.spyOn(fs, 'readFile').mockRejectedValue(new Error('Test readFile error'));
+
+        await expect(checkMarkdownFiles(tempDir, validateBlogs)).rejects.toThrow('Test readFile error');
+        expect(mockConsoleError).toHaveBeenCalledWith(
+            expect.stringContaining(`Error processing file invalid.md:`),
+            expect.any(Error)
+        );
+
+        mockReadFile.mockRestore();
+    });
+
+    it('should handle main function errors and exit with status 1', async () => {
+        jest.spyOn(fs, 'readdir').mockRejectedValue(new Error('Test error'));
+
+        await main();
+
+        expect(mockProcessExit).toHaveBeenCalledWith(1);
+
+        expect(mockConsoleError).toHaveBeenCalledWith(
+            'Failed to validate markdown files:',
+            expect.any(Error)
+        );
+    });
+
+    it('should handle successful main function execution', async () => {
+
+        await main();
+
+        expect(mockConsoleError).not.toHaveBeenCalledWith()
+    });
```

Comment:
not applying this one because we are essentially calling the checkMarkdownFiles function inside main, and we've already tested that in another test. So, in this case, we're just calling the main function to ensure it doesn't throw any errors. We have another test for main function errors.